printer.go

```diff
@@ -35,113 +47,58 @@ func NewPrinter() *Printer {
 		ValueInterval:   500 * time.Millisecond,
 	}
 }
-
-const nPrinterOut = len(printerHeadings)
-
-var (
-	printerHeadings = [...]string{
-		"Iter",
-		"FunEval",
-		"Obj",
-		"GradNorm",
-	}
-)
-
-func (p *Printer) Init(f *FunctionInfo) error {
-	p.printGrad = f.IsFunctionGradient || f.IsGradient
-
-	p.lastHeading = p.HeadingInterval + 1          // So the headings are printed the first time
-	p.lastValue = time.Now().Add(-p.ValueInterval) // So the values are printed the first time
+func (p *Printer) Init() error {
+	p.lastHeading = p.HeadingInterval
```

Comment:
I would keep the comments here. Particularly the second line is very weird if you don't understand the purpose.